src/observable/fromArrayLike.ts

```diff
@@ -0,0 +1,69 @@
+import {Scheduler} from '../Scheduler';
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+
+export class ArrayLikeObservable<T> extends Observable<T> {
+
+  private _mapFn: (x: number, y: any) => T;
+
+  static create<T>(arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, scheduler?: Scheduler) {
```

Comment:
the original does bind the thisArg though, but not sure if this is something people *really* care about. https://github.com/Reactive-Extensions/RxJS/blob/master/src/core/linq/observable/from.js#L101